Jenkinsfile

```diff
@@ -49,16 +40,29 @@ pipeline {
         sh 'git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master'
         sh 'git fetch --no-tags'
 
-        // Don't fail the whole build if octocatalog-diff fails, since it's new
-        // and needs some fixing before it's relied on
         script {
-          try {
-            sh 'make all_diffs'
-          } catch (err) {
-            echo 'make all_diffs failed, but it is being ignored for now'
-            mail to: 'jvperrin@ocf.berkeley.edu',
-                 subject: "all_diffs failed on ${JOB_NAME}/#${BUILD_NUMBER}",
-                 body: BUILD_URL
+          // This should only run for pull requests, so that it is able to post
+          // change/failure comments on the review
+          if (env.CHANGE_ID) {
```

Comment:
I believe so, but I haven't tested this explicitly since I've only used this branch so far for testing. From https://jenkins.io/doc/book/pipeline/multibranch/#additional-environment-variables it appears to be the pull request ID, so it should exist for non-fork-based PRs too.

Reply:
I believe so, but I haven't tested this explicitly since I've only used this branch so far for testing. From https://jenkins.io/doc/book/pipeline/multibranch/#additional-environment-variables it appears to be the pull request ID, so it should exist for non-fork-based PRs too.

Reply:
I think that for non-fork PRs our Jenkins does not run PR tests - it just does the standard branch tests. (Compare Jenkins runs for #821 and #822.) Maybe we should change that?

Reply:
Ah, I see what you mean, interesting. I'm not sure how to make the branches use the same workflow as pull requests, but the jenkins config for `puppet` does have an option for "Only branches that are also filed as PRs" instead of the current setting of "All branches". I've tried switching it to that and we'll see if that makes a difference or not.

Reply:
Yeah I don't think it makes a difference apart from filtering which branches have tests run, but maybe they'll have the right environment variable set then, that part still need to be tested.

Reply:
Oh never mind there's a "Discover pull requests from origin" option that makes a lot more sense, I'll try that instead.

Reply:
Yeah, this does what we want I believe, so branches that don't have a PR will build and skip this step, but branches that do have a PR or ones that are from a fork should use this workflow and post a comment.